### ISSUE TITLE
change NNUE file ?

### ISSUE THREAD

**Issue by tissatussa** [bitArtisan1/Wize#1]
I can compile your source on Xubuntu 22.04 and i wonder : can i change the NN file and compile ? Will your search / eval logic work with any another NN ? (being the same format)

**Comment by serptail** [comment]
Yes, it should work with another .nnue file, make sure to change the file name to your new .nnue in my **init.cpp** file.

**Comment by tissatussa** [comment]
@bitArtisan1 tnx, i will experiment with different compiles .. in README you state the "NNUE Evaluation" is "Features=HalfKP(Friend)[41024->256x2],Network=AffineTransform[1<-32](ClippedReLU[32](AffineTransform[32<-32](ClippedReLU[32](AffineTransform[32<-512](InputSlice[512(0:512)])))))" which means the good-old Stockfish 20 Mb ones .. lots of them exist, i once found "red neuronal", from some "Fischermann" or so, he trained many NN and maybe still does .. is your NNUE file unique ?

**Comment by serptail** [comment]
No problem! No, it's not unique—it's a pre-trained network from the Stockfish repositories, and I haven't made any modifications to it. You can find it in the Stockfish NNUE database [here.](https://tests.stockfishchess.org/nns?network_name=nn-9931db908a9b.nnue&user=)
Good luck with your experiments!

**Comment by tissatussa** [comment]
@bitArtisan1 yes, that SF db is well-known .. in the meantime i compiled Wize with several such 20 Mb NNs and let those engines play against eachother and also other NN-based engines of equal strength or having a (own) small NN (say upto 5 Mb) and see how they perform .. i used this position :

![e4-c6-d4-d6](https://github.com/user-attachments/assets/2f0dbe60-c83b-4518-b8c5-b0b1df6e50f5)

1. e4 c6 2. d4 d6!

Black plays the Caro-Kann Defense and should do 2...d5 but i'm experimenting with this d6-setup in my own games. SF often plays d6-d5! soon, though wasting a tempo .. pawns can not go back ! White should be able to win such game, but with many pieces on the board it's all about good moves!

soon i will post the game list here - have to prepare - but here's one nice game : your default NN wins with Black against Wize-with-Mida-NN (see https://github.com/GiacomoPorpiglia/Mida ) .. but Mida itself beats default-Wize, so i keep wondering what happens to the strength of any NN, all trained differently, in your Wize ?

![wize-demo-ApronusDiagram1735598274](https://github.com/user-attachments/assets/ef89c6e8-ddd8-4cf3-b3a2-06d1737de04e)

[Event "engine vs engine"]
[Site "Holland"]
[Date "2024.12.27"]
[Round "?"]
[White "Wize v1.0 NNUE mida"]
[Black "Wize v1.0 NNUE default"]
[Result "0-1"]
[FEN "rnbqkbnr/pp2pppp/2pp4/8/3PP3/8/PPP2PPP/RNBQKBNR w KQkq - 0 1"]
[GameDuration "00:13:34"]
[PlyCount "114"]
[SetUp "1"]
[TimeControl "300+3"]

{ https://www.lichess.org/r4UEkWWe } 1. Nf3 {+1.33/11 10s} Nf6 {-1.29/12 14s} 2. Nc3 {+1.65/11 8.3s}
g6 {-1.30/12 13s} 3. h3 {+1.70/11 9.0s} Bg7 {-1.16/12 22s} 4. Bd3 {+1.53/10 17s}
Nbd7 {-0.99/11 13s} 5. Be3 {+1.23/9 8.8s} e5 {-1.00/10 10s}
6. dxe5 {+1.12/10 15s} Nxe5 {-0.94/11 10s} 7. Nxe5 {+1.15/10 11s}
dxe5 {-0.94/11 11s} 8. a4 {+1.15/9 23s} Be6 {-0.94/10 21s} 9. a5 {+0.83/10 13s}
O-O {-0.84/11 17s} 10. Be2 {+0.92/9 7.5s} Qc7 {-0.44/10 7.5s}
11. O-O {+0.63/9 6.8s} b5 {-0.30/11 13s} 12. b4 {+0.26/9 13s} Rad8 {+0.50/9 14s}
13. Qc1 {+0.39/8 7.5s} Rd4 {+0.63/8 6.8s} 14. Rb1 {+0.19/8 15s}
Rfd8 {+0.35/8 6.5s} 15. a6 {+0.31/8 11s} Bf8 {+0.85/8 7.1s}
16. Bxb5 {-0.63/8 9.1s} cxb5 {+2.95/10 9.1s} 17. Nxb5 {-1.49/11 7.4s}
Qe7 {+2.72/10 5.6s} 18. Nxa7 {-1.68/10 6.8s} Qxa7 {+4.62/11 5.5s}
19. c3 {-5.64/10 11s} Nxe4 {+6.11/11 4.6s} 20. cxd4 {-6.17/10 4.8s}
exd4 {+7.37/12 11s} 21. Bh6 {-7.01/10 15s} Nc3 {+8.57/11 6.2s}
22. Rb2 {-8.63/11 13s} Qxa6 {+9.08/11 6.0s} 23. Bxf8 {-9.06/10 5.4s}
Rxf8 {+9.37/11 6.0s} 24. Qd2 {-8.31/9 4.4s} Bc4 {+10.02/12 6.5s}
25. Re1 {-8.73/11 4.2s} Qf6 {+10.95/12 6.6s} 26. Rc2 {-9.68/11 7.0s}
Kg7 {+11.05/11 13s} 27. f4 {-9.49/11 4.5s} Ra8 {+11.00/11 4.5s}
28. Kh2 {-9.01/10 5.0s} Rd8 {+10.69/11 5.0s} 29. Re5 {-9.16/11 5.9s}
Rd5 {+11.65/12 6.9s} 30. Re8 {-10.54/10 3.9s} Rf5 {+11.15/11 5.4s}
31. g3 {-9.91/11 3.9s} h5 {+11.39/11 6.8s} 32. h4 {-10.67/10 4.3s}
Rb5 {+12.07/10 11s} 33. Rc1 {-10.33/10 9.9s} Rxb4 {+11.63/10 3.6s}
34. f5 {-10.70/10 8.7s} gxf5 {+12.20/10 4.2s} 35. Rce1 {-11.94/9 4.3s}
f4 {+14.67/10 4.2s} 36. Qf2 {-13.49/9 3.7s} Nd5 {+18.45/10 7.0s}
37. gxf4 {-14.85/9 5.6s} Qxf4+ {+20.74/11 3.5s} 38. Qxf4 {-14.77/11 3.2s}
Nxf4 {+20.88/13 4.0s} 39. Rd8 {-14.82/12 4.5s} Ne6 {+20.58/12 3.2s}
40. Rd6 {-14.80/11 4.0s} Kf6 {+21.16/12 4.7s} 41. Kg3 {-15.65/12 5.5s}
Kf5 {+21.97/12 2.9s} 42. Rd7 {-16.49/12 4.5s} Rb3+ {+21.53/12 3.1s}
43. Kh2 {-16.71/12 9.8s} f6 {+22.34/12 4.6s} 44. Rc1 {-16.37/11 4.4s}
Rc3 {+22.33/12 3.4s} 45. Re1 {-18.68/11 3.3s} Kf4 {+22.95/13 5.8s}
46. Kh1 {-19.40/12 5.2s} f5 {+24.21/12 3.8s} 47. Rd1 {-19.61/12 3.2s}
Be2 {+26.48/11 2.9s} 48. Ra1 {-18.76/11 3.0s} Kg3 {+26.48/11 2.8s}
49. Kg1 {-21.45/11 2.8s} Nf4 {+27.01/11 3.9s} 50. Rg7+ {-24.58/12 3.4s}
Bg4 {+27.46/10 2.7s} 51. Rga7 {-26.97/13 6.4s} Rc2 {+27.76/13 6.4s}
52. R7a2 {-M12/14 5.0s} Be2 {+M11/17 2.7s} 53. Kh1 {-M10/20 2.8s}
Bf3+ {+M9/39 2.7s} 54. Kg1 {-M8/65 8.5s} Bg2 {+M7/65 8.8s}
55. Ra3+ {-M6/24 2.8s} Rc3 {+M5/24 3.9s} 56. Rxc3+ {-M4/44 8.1s}
dxc3 {+M3/128 0.029s} 57. Rd1 {-M2/128 0.027s}
Ne2# {+M1/128 0.015s, Black mates} 0-1

**Comment by tissatussa** [comment]
Here's the game list, 22 games 5m+3s.
ZIPped PGN : [Wize-v1.0-NNUE-multi.zip](https://github.com/user-attachments/files/18361644/Wize-v1.0-NNUE-multi.zip)

![Wize-game-results](https://github.com/user-attachments/assets/92511b20-302a-4b9f-b922-84a65bb136a3)

[ i doubt this can be of any help to improve the engine - HTH ]
